data/xml/schema.rnc

```diff
@@ -11,26 +11,36 @@ first = element first { text }
 last = element last { text }
 Person = attribute id { xsd:NCName }?, (first? & last)
 
+checksum = attribute hash { xsd:string {minLength="8" maxLength="8" pattern="[0-9a-f]*"} }
+url-with-checksum = element url {
+    xsd:anyURI {pattern="https?://.*"} |
+    (xsd:anyURI {pattern=".*"}, checksum)
```

Comment:
Why did you add the .* pattern here? If it really does something useful, maybe a short comment would be useful.  Restricting by .* seems to be contradictory otherwise.

Reply:
Ah, I started from your suggestion which included `pattern="/.*"`, but the slash didn't make sense so I removed it and left it like that without thinking further. You're right that it's superfluous.

Reply:
I would go back to "/.*" to denote that it can only be local.  If that was not obvious beforehand, maye add a comment à la:
```
    // We do not control the content of external URLS, therefore they have no checksums
    xsd:anyURI {pattern="https?://.*"} |
    // All data hosted by us (with relative paths, i.e. starting with /) need to have a checksum
    (xsd:anyURI {pattern="/.*"}, checksum)
```